micropython/ssa/core.py

```
@@ -157,25 +156,57 @@ def __connect(self, last_will: str | None = None, _with_registration: bool = Fal
+    def __handle_user_config(self, config: str):
         raise Exception(f"[TODO] ssa.__handle_config_change not implemented")
 
+    def register_handler(self, task: function):
+        """! Register a task to be executed as part of the main loop
+            @param task: The task to be executed
+                         The task should be an async function decorated with @ssa_property_handler or @ssa_event_handler
+                         See the ssa.decorators.py documentation for more information
+        """
+        self.__tasks.append(asyncio.create_task(task()))
+
+    def action_callback(self, action: str, callback_func: funtion, qos: int = 0):
```

Ruff (0.8.2)

Undefined name `funtion`

(F821)